### ISSUE TITLE
Unexpected error in logs while reporting failure message

### ISSUE THREAD

**Issue by cpinjani** [rancher/eks-operator#541]
**Rancher version:**
Rancher version: v2.9-989139e96a9de1b1792293e75d4e26e694f9a66d-head
eks-operator: rancher-eks-operator:104.0.0+up1.9.0-rc.1

**Cluster Type:** Downstream EKS cluster

**Describe the bug:**
Unexpected error in logs while reporting failure message

**Logs:**
**time="2024-05-20T07:08:54Z" level=error msg="Error removing metadata from failure message: message body not formatted as expected"**
time="2024-05-20T07:08:54Z" level=error msg="error syncing 'cattle-global-data/c-f2t4w': handler eks-controller: node group names must be unique within the [c-f2t4w] cluster to avoid duplication, requeuing"

**PR's**
- [x] https://github.com/rancher/eks-operator/pull/563
- [x] https://github.com/rancher/eks-operator/pull/564

**Comment by cpinjani** [comment]
Validated as fixed on `eks-operator:v1.9.0-rc.5`, the reported error message is not occurring the logs.
```
time="2024-06-09T14:29:53Z" level=error msg="error syncing 'cattle-global-data/c-j2qkm': handler eks-controller: node group names must be unique within the [c-j2qkm] cluster to avoid duplication, requeuing"
time="2024-06-09T14:29:53Z" level=error msg="Error recording ekscc [] failure message: resource name may not be empty"
time="2024-06-09T14:29:53Z" level=error msg="error syncing 'cattle-global-data/c-j2qkm': handler eks-controller: node group names must be unique within the [c-j2qkm] cluster to avoid duplication, requeuing"
time="2024-06-09T14:29:53Z" level=error msg="error syncing 'cattle-global-data/c-j2qkm': handler eks-controller: node group names must be unique within the [c-j2qkm] cluster to avoid duplication, requeuing"
time="2024-06-09T14:29:53Z" level=error msg="error syncing 'cattle-global-data/c-j2qkm': handler eks-controller: node group names must be unique within the [c-j2qkm] cluster to avoid duplication, requeuing"
time="2024-06-09T14:29:53Z" level=error msg="error syncing 'cattle-global-data/c-j2qkm': handler eks-controller: node group names must be unique within the [c-j2qkm] cluster to avoid duplication, requeuing"
```

However, seeing another error messages logs for some failure operations:
- time="2024-06-09T14:29:53Z" level=error msg="Error recording ekscc [] failure message: resource name may not be empty"
- time="2024-06-09T15:34:17Z" level=error msg="Error recording ekscc [] failure message: Operation cannot be fulfilled on eksclusterconfigs.eks.cattle.io \"c-hhwnb\": the object has been modified; please apply your changes to the latest version and try again"

@mjura Is it something we can fix in this issue ?

**Comment by mjura** [comment]
> @mjura Is it something we can fix in this issue ?

It is another issue, please open another card
